test/e2e/helpers.go

```diff
@@ -858,29 +859,39 @@ func getPodLogs(ctx context.Context, clientset *kubernetes.Clientset, pod corev1
 }
 
 // InstallHelmChart takes a helm repo URL, a chart name, and release name, and installs a helm release onto the E2E workload cluster.
-func InstallHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, repoURL, chartName, releaseName string, values []string) {
+func InstallHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, namespace, repoURL, chartName, releaseName string, options *helmVals.Options) {
 	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+	clientSet := clusterProxy.GetClientSet()
+
+	// create namespace if it doesn't already exist.
+	if _, err := e2e_namespace.Get(ctx, clientSet, namespace); err != nil {
+		if !apierrors.IsNotFound(err) {
+			Expect(err).NotTo(HaveOccurred())
+		}
+		if _, err := e2e_namespace.Create(ctx, clientSet, namespace, nil); err != nil {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+
 	kubeConfigPath := clusterProxy.GetKubeconfigPath()
 	settings := helmCli.New()
 	settings.KubeConfig = kubeConfigPath
 	actionConfig := new(helmAction.Configuration)
-	ns := "default"
-	err := actionConfig.Init(settings.RESTClientGetter(), ns, "secret", Logf)
+	err := actionConfig.Init(settings.RESTClientGetter(), namespace, "secret", Logf)
 	Expect(err).To(BeNil())
 	i := helmAction.NewInstall(actionConfig)
 	if repoURL != "" {
 		i.RepoURL = repoURL
 	}
 	i.ReleaseName = releaseName
-	i.Namespace = ns
+	i.Namespace = namespace
```

Comment:
The Helm install client has a flag to create a namespace if it doesn't exist. You can add this below.
```
i.CreateNamespace = true
```

Reply:
thank you, I didn't know, that's great!!